opcua-core/src/errors.rs

```diff
@@ -0,0 +1,17 @@
+// OPCUA for Rust
+// SPDX-License-Identifier: MPL-2.0
+
+//!  Rust OpcUa specific errors
+
+use opcua_types::VariantScalarTypeId;
+use thiserror::Error;
+
+/// Rust OpcUa specific errors
+#[allow(missing_docs)]
+#[derive(Error, Debug)]
+pub enum OpcUAError {
+    #[error("Received an expected variant type")]
+    UnExpectedVariantType(Option<VariantScalarTypeId>),
```

Comment:
`Unexpected` is one word, so the PascalCase version of this variant should be:
```suggestion
    UnexpectedVariantType(Option<VariantScalarTypeId>),
```